code/modules/projectiles/ammunition.dm

```diff
@@ -398,27 +447,110 @@ Turn() or Shift() as there is virtually no overhead. ~N
 	new /obj/item/stack/sheet/cardboard(T)
 	qdel(src)
 
-/obj/item/ammo_box/magazine/proc/deploy_ammo_box(mob/living/user, turf/T)
-	for(var/obj/structure/magazine_box/MB in T.contents)
-		to_chat(user, SPAN_WARNING("There is a [MB] deployed here already."))
+/obj/item/ammo_box/magazine/proc/deploy_ammo_box(var/mob/living/user, var/turf/T)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
 		return
+
+	var/box_on_tile = 0
+	for(var/obj/structure/magazine_box/found_MB in T.contents)
+		if(limit_per_tile != found_MB.limit_per_tile)
+			to_chat(user, SPAN_WARNING("You can't deploy different size boxes in one place."))
+			return
+		box_on_tile++
+		if(box_on_tile >= limit_per_tile)
+			to_chat(user, SPAN_WARNING("There is a maximum possible amount of boxes deployed here already."))
+			return
+
 	var/obj/structure/magazine_box/M = new /obj/structure/magazine_box(T)
 	M.icon_state = icon_state_deployed ? icon_state_deployed : icon_state
 	M.name = name
 	M.desc = desc
 	M.item_box = src
+	M.can_explode = can_explode
+	M.limit_per_tile = limit_per_tile
 	M.update_icon()
+	if(limit_per_tile > 1)
+		M.assign_offsets(T)
 	user.drop_inv_item_on_ground(src)
 	Move(M)
 
 /obj/item/ammo_box/magazine/afterattack(atom/target, mob/living/user, proximity)
+	if(burning)
+		to_chat(user, SPAN_DANGER("It's on fire and can explode!"))
+		return
 	if(!proximity)
 		return
 	if(isturf(target))
 		var/turf/T = target
 		if(!T.density)
 			deploy_ammo_box(user, T)
 
+//---------------------FIRE HANDLING PROCS
+
+/obj/item/ammo_box/magazine/flamer_fire_act(var/damage, var/datum/cause_data/flame_cause_data)
```

Comment:
this is full of duped code from /ammo_box/flamer_fire_act() and needs to be separated out into procs on /ammo_box (such as for getting severity, and for igniting)